native/Avalonia.Native/src/OSX/menu.mm

```diff
@@ -65,49 +76,77 @@ - (void)didSelectItem:(nullable id)sender
     return _native;
 }
 
-HRESULT AvnAppMenuItem::SetSubMenu (IAvnAppMenu* menu)
+HRESULT AvnAppMenuItem::SetSubMenu (IAvnMenu* menu)
 {
-    auto nsMenu = dynamic_cast<AvnAppMenu*>(menu)->GetNative();
-    
-    [_native setSubmenu: nsMenu];
-    
-    return S_OK;
+    @autoreleasepool
+    {
+        if(menu != nullptr)
+        {
+            auto nsMenu = dynamic_cast<AvnAppMenu*>(menu)->GetNative();
+            
+            [_native setSubmenu: nsMenu];
+        }
+        else
+        {
+            [_native setSubmenu: nullptr];
+        }
+        
+        return S_OK;
+    }
 }
 
 HRESULT AvnAppMenuItem::SetTitle (void* utf8String)
 {
-    if (utf8String != nullptr)
+    @autoreleasepool
     {
-        [_native setTitle:[NSString stringWithUTF8String:(const char*)utf8String]];
+        if (utf8String != nullptr)
+        {
+            [_native setTitle:[NSString stringWithUTF8String:(const char*)utf8String]];
+        }
+        
+        return S_OK;
     }
-    
-    return S_OK;
 }
 
 HRESULT AvnAppMenuItem::SetGesture (void* key, AvnInputModifiers modifiers)
 {
-    NSEventModifierFlags flags = 0;
-    
-    if (modifiers & Control)
-        flags |= NSEventModifierFlagControl;
-    if (modifiers & Shift)
-        flags |= NSEventModifierFlagShift;
-    if (modifiers & Alt)
-        flags |= NSEventModifierFlagOption;
-    if (modifiers & Windows)
-        flags |= NSEventModifierFlagCommand;
-    
-    [_native setKeyEquivalent:[NSString stringWithUTF8String:(const char*)key]];
-    [_native setKeyEquivalentModifierMask:flags];
-    
-    return S_OK;
+    @autoreleasepool
+    {
+        NSEventModifierFlags flags = 0;
+        
+        if (modifiers & Control)
+            flags |= NSEventModifierFlagControl;
+        if (modifiers & Shift)
+            flags |= NSEventModifierFlagShift;
+        if (modifiers & Alt)
+            flags |= NSEventModifierFlagOption;
+        if (modifiers & Windows)
+            flags |= NSEventModifierFlagCommand;
+        
+        [_native setKeyEquivalent:[NSString stringWithUTF8String:(const char*)key]];
+        [_native setKeyEquivalentModifierMask:flags];
+        
+        return S_OK;
+    }
 }
 
 HRESULT AvnAppMenuItem::SetAction (IAvnPredicateCallback* predicate, IAvnActionCallback* callback)
 {
-    _predicate = predicate;
-    _callback = callback;
-    return S_OK;
+    @autoreleasepool
```

Comment:
We should probably adopt Chrome's model with managing pools via RAII and skipping pool initialization if one is present. The required machinery is already there in app.mm (at least in native embedding branch)

Definitely not in the scope of this PR though

Reply:
+